src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+            EnFz_BeginWaitingPreFreeze(this);
+        }
+    }
+}
+
+void EnFz_BeginWaitingPreFreeze(EnFz* this) {
+    this->state = 1;
+    this->timer = 40;
+    this->isAlwaysTrue = true;
+    this->isFreezing = true;
+    this->actor.flags |= 1;
+    this->actionFunc = EnFz_WaitPreFreeze;
+    this->actor.gravity = -1.0f;
+}
+
+void EnFz_WaitPreFreeze(EnFz* this, GlobalContext* globalCtx) {
+    EnFz_RotateTowardsLink(this);
+    if (this->timer == 0) {
+        EnFz_MoveTowardsLink(this);
+    }
+}
+
+void EnFz_MoveTowardsLink(EnFz* this) {
+    this->state = 1;
+    this->isMoving = true;
+    this->timer = 100;
+    this->actionFunc = EnFz_Stop;
+    this->xzSpeed = 4.0f;
+}
+
+void EnFz_Stop(EnFz* this, GlobalContext* globalCtx) {
+    if ((this->timer == 0) || (this->isMoving == 0)) {
+        EnFz_BeginWaitingPostFreeze(this);
+    }
+}
+
+void EnFz_BeginWaitingPostFreeze(EnFz* this) {
+    this->state = 1;
+    this->timer = 40;
+    this->actionFunc = EnFz_WaitPostFreeze;
+    this->xzSpeed = 0.0f;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnFz_WaitPostFreeze(EnFz* this, GlobalContext* globalCtx) {
+    EnFz_RotateTowardsLink(this);
+    if (this->timer == 0) {
+        EnFz_BeginFreezing(this, globalCtx);
+    }
+}
+
+void EnFz_BeginFreezing(EnFz* this, GlobalContext* globalCtx) {
+    this->state = 1;
+    this->timer = 80;
+    this->actionFunc = EnFz_Freezing;
+    EnFz_GetCollision(this, globalCtx);
+}
+
+void EnFz_Freezing(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f vec1;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 isTimerMod8;
+    s16 primAlpha;
+
+    if (this->timer == 0) {
+        EnFz_BeingShrinking(this);
+    } else if (this->timer >= 11) {
+
+        isTimerMod8 = false;
+        primAlpha = 150;
+        func_8002F974(&this->actor, NA_SE_EN_FREEZAD_BREATH - SFX_FLAG);
+
+        if ((this->timer - 10) < 16) {            // t < 26 
+            primAlpha = (this->timer * 10) - 100; 
+        }
+
+        accel.z = 0.0f;
+        accel.x = 0.0f;
+        accel.y = 0.6f;
+
+        pos.x = this->actor.posRot.pos.x;
+        pos.y = this->actor.posRot.pos.y + 20.0f;
+        pos.z = this->actor.posRot.pos.z;
+
+        Matrix_RotateY((this->actor.shape.rot.y / 32768.0f) * M_PI, MTXMODE_NEW);
+
+        vec1.x = 0.0f;
+        vec1.y = -2.0f;
+        vec1.z = 20.0f; // xz velocity
+        Matrix_MultVec3f(&vec1, &velocity);
+
+        if ((this->timer % 8) == 0) {
+            isTimerMod8 = true;
+        }
+
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, isTimerMod8);
+        pos.x += (velocity.x * 0.5f);
+        pos.y += (velocity.y * 0.5f);
+        pos.z += (velocity.z * 0.5f);
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, false);
+    }
+}
+
+void EnFz_BeginDespawn(EnFz* this, GlobalContext* globalCtx) {
```

Comment:
These types of functions that are called to set to do some setup and then set the action function to `ActorName_Foo` are called `ActorName_SetupFoo`
```suggestion
void EnFz_SetupDespawn(EnFz* this, GlobalContext* globalCtx) {
```